lib/array_equals.rb

```diff
@@ -1,5 +1,25 @@
-# Determines if the two input arrays have the same count of elements
-# and the same integer values in the same exact order
 def array_equals(array1, array2)
+  if array1 == nil || array2 == nil || array1 == [] || array2 == []
+    if array1 == array2
+      return true
+    else
+      return false
+    end
+  end
+
+  if array1.length != array2.length
+    return false
+  else
+    identical_array = false
+    for index in array1
+      if array1 == array2
+        identical_array = true
+      else
+        identical_array = false
+        return identical_array
+      end
+    end
+    return identical_array
```

Comment:
You can just "return true" here. No need to save into the identical_array variable